controller/engine_controller.go

```diff
@@ -254,7 +254,9 @@ func (ec *EngineController) syncEngine(key string) (err error) {
 		if engine.Status.InstanceManagerName != "" {
 			// don't go through state transition because it can go wrong
 			if _, err := ec.DeleteInstance(engine); err != nil {
-				return err
+				if !strings.Contains(err.Error(), "cannot find") {
```

Comment:
We need to properly define the error type instead of relying on `cannot find` keyword in case keyword changed or become CamelCase. If the error message is coming from executable we have no choices, but if it's still in the process we should be able to use a more reliable way.

Reply:
Maybe we can use Kubernetes's apierrors package.

Reply:
In fact, better not to use Kubernetes's one. We can define something simple ourselves and involve from there. Because it's not strictly API layer errors.

Reply:
E.g.
```
type Error struct {
    error string
    reason ErrorReason
}

func NewError(error string, reason ErrorReason) *Error {
    return xxxx
}

func (e *Error) IsNotFound bool (
    return e.ErrorReason == ErrorReasonNotFound
)

Reply:
This needs to be shared in all longhorn repos. Hence it's better to put it into `longhorn-instance-manager/types/`

Reply:
ok, this is troublesome.

Basically what we need is a defined error for all the longhorn components, since we are going to have manager talk to the engine directly as well. And it's weird to refer to e.g. `imerrors` for the call we made to the engine controller.

Before we have a separate repo for the api machinery, let's add `types.ErrorIsNotFound()` and use it to cover the error checking for now.

Reply:
OK